### PR TITLE
[ML][PYTORCH] Update macOS build to libtorch 1.9

### PR DESCRIPTION
**Pull request by davidkyle**
Changes the BLAS library to be explicitly set to `vecLib` which is part of the `Accelerate` framework.

non-issue as the change is to unreleased code

### CODE REVIEW

File: dev-tools/download_macos_deps.sh [hunk 1]
```diff
@@ -25,7 +25,7 @@ DEST=/usr
 case `uname -m` in
 
     arm64)
-        ARCHIVE=local-arm64-apple-macosx11.1-3.tar.bz2
+        ARCHIVE=local-arm64-apple-macosx11.1-4.tar.bz2
```
Comment (droberts195) [inline dev-tools/download_macos_deps.sh:28]: This file is in place now.

File: build-setup/macos.md [hunk 2]
```diff
@@ -183,18 +183,18 @@ external processes.
...
 export BUILD_CAFFE2=OFF
 export USE_NUMPY=OFF
 export USE_DISTRIBUTED=OFF
-export USE_MKLDNN=OFF
+export USE_MKLDNN=ON
 export USE_QNNPACK=OFF
 export USE_PYTORCH_QNNPACK=OFF
 [ $(uname -m) = x86_64 ] && export USE_XNNPACK=OFF
 # TODO: recheck if this is still necessary next time we upgrade
 [ $(uname -m) != x86_64 ] && export CMAKE_OSX_ARCHITECTURES=`uname -m`
```
Comment (droberts195) [inline build-setup/macos.md:196]: These two lines can be deleted now - the workaround is no longer necessary.

**Review by droberts195 — APPROVED** [review]
LGTM

There is one more workaround that can be removed for ARM.

File: build-setup/macos.md [hunk 3]
```diff
@@ -183,18 +183,18 @@ external processes.
...
 
 ```
-# TODO: add BLAS=vecLib when we upgrade to 1.9
+export BLAS=vecLib
 export BUILD_TEST=OFF
 export BUILD_CAFFE2=OFF
 export USE_NUMPY=OFF
 export USE_DISTRIBUTED=OFF
-export USE_MKLDNN=OFF
+export USE_MKLDNN=ON
```
Comment (droberts195) [inline build-setup/macos.md:191]: Turns out this doesn't work in PyTorch 1.9 when built for ARM macOS.

The fixes that make `oneDNN` work on Apple silicon are in versions 2.1.3 and 2.2 according to https://github.com/oneapi-src/oneDNN/issues/984#issuecomment-811952961.  And PyTorch 1.9 is using version 2.1.2 😿 .

So I'll make this setting conditional on the CPU type for the time being.

Comment (droberts195) [reply build-setup/macos.md]: It should be possible to build MKLDNN on M1 in PyTorch 1.10 - see https://github.com/pytorch/pytorch/pull/57928.  We'll have to see if there are further problems downstream of simply building it though.

File: dev-tools/download_macos_deps.sh (hunk 1 above)
Comment (droberts195) [reply dev-tools/download_macos_deps.sh:28]: This file is now updated to a version containing libraries for the correct architecture. 🤦

**Comment by davidkyle** [comment]
retest

**Pull request merged**